### DISCUSSION TITLE
how to change scene correctly

### DISCUSSION THREAD

**Discussion by ghost** [stride3d/stride#1392] [Q&A]
i'm currently changing scenes by just setting the root scene to the one i want to change to. this works fine, until i want to switch to a scene that i previously loaded. it changes to the scene as it was when i left it, meaning the scenes stay in ram after i've changed to a different one. i'm not sure how to unload the scenes, as they are loaded by the content manager in the previous scene, so i can't call content. unload on it after switching scenes. i'm not sure how to fix this.

**Comment by manio143** [comment]
Not sure why you can't call `Content.Unload` on the previous scene - that's exactly what you want to do, sounds like.

**Reply by ghost** [reply]
is crashes with "resource was not loaded with this content manager" or something similar. i loaded it my calling content.load in the previous scene though.
